src/ArcxAnalyticsSdk.ts

```diff
@@ -41,32 +53,30 @@ export class ArcxAnalyticsSdk {
       this._trackPagesChange()
     }
 
-    if (sdkConfig.trackWalletConnections) {
-      this._reportCurrentWallet()
-      window.ethereum?.on('accountsChanged', (...args: unknown[]) =>
-        this._onAccountsChanged(args[0] as string[]),
-      )
-    }
-
-    if (sdkConfig.trackChainChanges) {
-      window.ethereum?.on('chainChanged', (...args: unknown[]) =>
-        this._onChainChanged(args[0] as string),
-      )
-    }
-
-    if (this.sdkConfig.trackTransactions) {
-      this._trackTransactions()
-    }
-
-    if (this.sdkConfig.trackSigning) {
-      this._trackSigning()
-    }
+    this._handleAccountDisconnected = this._handleAccountDisconnected.bind(this)
+    this.setProvider = this.setProvider.bind(this)
```

Comment:
These functions are used inside listeners and use `this.`, which is lost during the event call. For this reason, I'm binding `this` to them.